src/react/boosts/payments/BoostPayStripe.tsx

```diff
@@ -1,26 +1,106 @@
-import { BoostPayOption, BoostPayCardDetails } from "../BoostStyledComponents";
+import { Mutation, MutationFunc } from "react-apollo";
+import { boostPayStripeMutation } from "../queries";
+import makeAsyncScriptLoader from "react-async-script";
+import { BoostPayCardDetails, BoostFlexCenter, BoostButton } from "../BoostStyledComponents";
+import { StripeProvider, Elements } from "react-stripe-elements";
+import BoostPayFormStripe from "./BoostPayFormStripe";
+import { LoadingMessage } from "@joincivil/components";
+import { BoostPayOption } from "./BoostPayOption";
 
 export interface BoostPayStripeProps {
-  defaultChecked: boolean;
-  value: string;
+  boostId: string;
+  newsroomName: string;
+  usdToSpend: number;
+  selected: boolean;
+  paymentType: string;
+  optionLabel: string | JSX.Element;
+  paymentStarted?: boolean;
+  handleNext(): void;
+  handlePaymentSelected?(paymentType: string): void;
+  handlePaymentSuccess(): void;
 }
 
-export const BoostPayStripe: React.FunctionComponent<BoostPayStripeProps> = props => {
-  return (
-    <BoostPayOption>
-      <BoostPayRadioBtn value={props.value} defaultChecked={props.defaultChecked} disabled={true}>
-        Pay using card
-      </BoostPayRadioBtn>
-      <BoostPayCardDetails>
-        <p>
-          Coming Soon{" "}
-          <a href="#TODO" target="_blank">
-            Why only ETH?
-          </a>
-        </p>
-      </BoostPayCardDetails>
-    </BoostPayOption>
-  );
-};
+export interface BoostPayStripeStates {
+  stripeLoaded: boolean;
+  stripe: any;
+}
+
+export class BoostPayStripe extends React.Component<BoostPayStripeProps, BoostPayStripeStates> {
+  constructor(props: BoostPayStripeProps) {
+    super(props);
+    this.state = {
+      stripeLoaded: false,
+      stripe: null,
+    };
+  }
+
+  public render(): JSX.Element {
+    if (this.props.paymentStarted) {
+      return <>{this.renderPaymentForm()}</>;
+    }
+
+    return <>{this.renderDefaultOption()}</>;
+  }
+
+  private renderDefaultOption = (): JSX.Element => {
+    return (
+      <BoostPayOption
+        paymentType={this.props.paymentType}
+        optionLabel={this.props.optionLabel}
+        selected={this.props.selected}
+        handlePaymentSelected={this.props.handlePaymentSelected}
+      >
+        <BoostPayCardDetails>
+          <BoostFlexCenter>
+            <p>
+              Continue with adding your payment information. Your payment information will be processed through{" "}
+              <a href="https://stripe.com/" target="_blank">
+                Stripe
+              </a>
+              .
+            </p>
+            {this.props.selected && <BoostButton onClick={() => this.props.handleNext()}>Next</BoostButton>}
+          </BoostFlexCenter>
+        </BoostPayCardDetails>
+      </BoostPayOption>
+    );
+  };
+
+  private renderPaymentForm = (): JSX.Element => {
+    // TODO(sruddy) TEST API KEY >> add util for test key, prod keys and js
```

Comment:
You can get this via environment variable and importing `CivilContext` from `@joincivil/components` I think.

- Add the key for deployments via orchestration repo e.g. https://github.com/joincivil/orchestration/commit/0053f797b95eada235daea4a321f61ad0b506ca6
- Add test key to `packages/dapp/.env` for local usage
- Whole flow of consuming key from env into config into context into consumer in this commit: https://github.com/joincivil/Civil/commit/96fdd07d2e1c6151f6be87f45be0051db709bcc8